modules/storage/cosmos-db/main.bicep

```
@@ -1,99 +1,353 @@
+param enableAnalyticalStorage bool = false
+
+@description('The type of schema for analytical storage.')
+@allowed([ 'FullFidelity', 'WellDefined' ])
+param analyticalStorageSchemaType string = 'WellDefined'
+
+@description('The list of Cassandra keyspaces configurations with tables.')
+param cassandraKeyspaces array = []
+
+@description('The list of SQL databases configurations with containers, its triggers, storedProcedures and userDefinedFunctions.')
+param sqlDatabases array = []
+
+@description('The list of SQL role definitions.')
+param sqlRoleDefinitions array = []
+
+@description('The list of SQL role assignments.')
+param sqlRoleAssignments array = []
+
+@description('The list of MongoDB databases configurations with collections, its indexes, Shard Keys.')
+param mongodbDatabases array = []
+
+@description('The list of Gremlin databases configurations with graphs.')
+param gremlinDatabases array = []
+
+@description('The list of Table databases configurations.')
+param tables array = []
+
+@description('Array of role assignment objects that contain the "roleDefinitionIdOrName" and "principalId" to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, provide either the display name of the role definition, or its fully qualified ID in the following format: "/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11"')
+param roleAssignments array = []
+
+@description('The type of identity used for the Cosmos DB account. The type "SystemAssigned, UserAssigned" includes both an implicitly created identity and a set of user-assigned identities. The type "None" will remove any identities from the Cosmos DB account.')
+@allowed([ 'None', 'SystemAssigned', 'SystemAssigned,UserAssigned', 'UserAssigned' ])
+param identityType string = 'None'
+
+@description('The list of user-assigned managed identities. The user identity dictionary key references will be ARM resource ids in the form: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}"')
+param userAssignedIdentities object = {}
+
+@description('Private Endpoints that should be created for Azure Cosmos DB account.')
+param privateEndpoints array = []
+
+@description('List of key-value pairs that describe the resource.')
+param tags object = {}
+
+@allowed([
+  'CanNotDelete'
+  'NotSpecified'
+  'ReadOnly'
+])
+@description('Specify the type of lock on Cosmos DB account resource.')
+param lock string = 'NotSpecified'
+
+var varConsistencyPolicy = (defaultConsistencyLevel == 'BoundedStaleness') ? {
+  defaultConsistencyLevel: 'BoundedStaleness'
+  maxStalenessPrefix: maxStalenessPrefix
+  maxIntervalInSeconds: maxIntervalInSeconds
+} : {
+  defaultConsistencyLevel: defaultConsistencyLevel
 }
 
-var secondaryRegions = [for (region, i) in secondaryLocations: {
+var varSecondaryRegions = [for (region, i) in secondaryLocations: {
```

```suggestion
var _secondaryLocations = [for (region, i) in secondaryLocations: {
```

Having variables prefixed with var is a little redundant. How about dropping the prefixes, and in places where a var shadows a param, we can just use the prefix '_'.

---

Personally I'd prefer avoid using `_` as a prefix. Can we make the variable name more descriptive, e.g., `secondaryRegionsWithDefaults`?

---

Updated variable names with descriptive name as suggested.